subgraphs/playback/src/utils/mocks.ts

```diff
@@ -1,5 +1,190 @@
+      name: `My Computer`,
+      type: 'Computer',
+      is_active: true,
+      volume_percent: 100,
+    },
+    is_playing: true,
+    item: { id: '2jJIENqpTOjBECelzBJAVL' },
+    progress_ms: 0,
+    repeat_state: 'off',
+    shuffle_state: false,
+    timestamp: Date.now(),
+    currently_playing_type: 'track',
+  },
+};
+
+export function addUser(userId: string) {
+  console.log(`trying to add user ${userId}`);
+  if (userState[userId]) return;
+
+  //TODO - pick a song to use for playback state
+  userState[userId] = userState['default'];
+}
+
+export class MockedSpotifyDataSource implements SpotifyDataSource {
+  private userId: string;
+  constructor(userId: string) {
+    this.userId = userId;
+  }
+  getDevices(): Promise<Spotify.Object.List<'devices', Spotify.Object.Device>> {
+    return new Promise(() => ({
+      devices: [userState[this.userId]?.device],
+    }));
+  }
+  getPlaybackState(params?: {
+    additional_types?: string;
+  }): Promise<Spotify.Object.PlaybackState> {
+    if (userState[this.userId] && userState[this.userId].is_playing) {
+      if (userState[this.userId].progress_ms >= 140000)
+        userState[this.userId].progress_ms = 0;
+      else userState[this.userId].progress_ms += 1000;
+    }
+    return new Promise((resolve) => resolve(userState[this.userId] as any));
+  }
+  resumePlayback({
+    body,
+    params,
+  }: {
+    params: { device_id?: string };
+    body: {
+      context_uri?: string;
+      uris?: string[];
+      offset?: { position?: number; uri?: string };
+      position_ms?: number;
+    };
+  }): Promise<boolean> {
+    if (userState[this.userId] && !userState[this.userId].is_playing) {
+      userState[this.userId].is_playing = true;
+      userState[this.userId].device.is_active = true;
+    } else return new Promise((resolve) => resolve(false));
+
+    return new Promise((resolve) => resolve(true));
+  }
+  pausePlayback({
+    params,
+  }: {
+    params: { device_id?: string };
+  }): Promise<boolean> {
+    if (userState[this.userId] && userState[this.userId].is_playing) {
+      userState[this.userId].is_playing = false;
+      userState[this.userId].device.is_active = false;
+    } else return new Promise((resolve) => resolve(false));
+
+    return new Promise((resolve) => resolve(true));
+  }
+  seekToPosition({
+    params,
+  }: {
+    params: { position_ms: number; device_id?: string };
+  }): Promise<boolean> {
+    if (userState[this.userId])
+      userState[this.userId].progress_ms = params.position_ms;
+    else return new Promise((resolve) => resolve(false));
+
+    return new Promise((resolve) => resolve(true));
+  }
+  setRepeatMode({
+    params,
+  }: {
+    params: { state: Spotify.Object.RepeatMode; device_id?: string };
+  }): Promise<boolean> {
+    if (userState[this.userId])
+      userState[this.userId].repeat_state = params.state;
+    else return new Promise((resolve) => resolve(false));
+
+    return new Promise((resolve) => resolve(true));
```

Comment:
```suggestion
    else return Promise.resolve(false);

    return new Promise.resolve(true);
```

You can use the `Promise.resolve` helper here to shorten this a bit :)